### ISSUE TITLE
Previously fixed errors break watcher build

### ISSUE THREAD

**Issue by tysonnero** [webpack-contrib/stylelint-webpack-plugin#24]
I am experiencing an issue where I'm running a webpack watch, a stylelint error is reporting, I fix the error, then the webpack watcher is continually reporting errors. It seems like it's continuously stuck in error mode. My CSS config looks something like this. Any ideas?

```
  name: 'css',
  context: paths.css.entry,
  entry: cssFiles,
  output: {
    path: paths.css.output,
    filename: 'xxx/assets/css/[name].min.css'
  },
  module: {
    loaders: [
      { test: /\.scss/, loader: ExtractTextPlugin.extract('style', 'css!postcss!sass') },
      { test: /\.(woff2?|ttf|eot|svg)(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: 'file?name=/xxx/assets/fonts/[hash].[ext]' },
      { test: /\.(jpg|jpeg|gif|png|tiff)(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: 'file?name=/xxx/assets/images/[hash].[ext]' }
    ]
  },
  sassLoader: {
    // Set outputStyle: 'expanded' as the default (nested) has wrapping and spacing issues
    // that cause linting errors
    outputStyle: 'expanded'
  },
  plugins: [
    new styleLintPlugin({
      syntax: 'scss',
      failOnError: false,
    }),
    new ExtractTextPlugin('creditcards/assets/css/[name].min.css')
  ],
  postcss: [
    autoprefixer({ browsers: ['> 5% in US', 'last 2 versions', 'IE 9'] }),
    reporter({ clearMessages: true, throwError: true })
  ]
```

**Comment by laander** [comment]
Encountering the same (or closely related issue. 

If `failOnError: true` and im running webpack watch, it breaks silently on the first time an error occurs and just stays there. Even if I'm editing the file and fixes the error, nothing happens - it seems like the linting reports the error incorrectly and webpack aren't sure how to proceed. 

This doesn't happen with `failOnError: false` as webpack will just build whether theres an error or not (it does get reported though).

**Comment by laander** [comment]
At https://github.com/vieron/stylelint-webpack-plugin/blob/v0.4.0/lib/run-compilation.js#L34 - should the plugin really throw an exception and not just parse a string as parameter to `done`?

**Comment by zuanoc** [comment]
I have the same issue with the `failOnError: true`. The build watcher just breaks and can not recover.

```
app/sass/main.scss                                                                                                                        
 3:1  ×  Unexpected empty line before at-rule   at-rule-empty-line-before                                                                 
                                                                                                                                                                                                                                                                        
(node:7508) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Failed because of a stylelint error.  
```

**Comment by JaKXz** [comment]
@tysonnero @laander @zuanoc sorry for the delayed response, are you still seeing this issue with v0.7.0?

**Comment by shellscape** [comment]
Closing as stale due to age and inactivity. If there is still interest in this issue, please do submit a pull request.

**Comment by maxjf1** [comment]
Did anyone solve this? I want to alert about lint erros running watch mode. But either i enable `failOnError` and it just stop working on the first time, or i set it to false and it just don't lint :(
